kirin/cots/model_maker.py

```
@@ -518,10 +520,24 @@ def _make_trip_update(self, json_train, vj, action_on_trip=ActionOnTrip.NOT_ADDE
                         )
                     )
 
-                arr_dep_status = getattr(
+                stop_event_status = getattr(
                     st_update, _status_map[arrival_departure_toggle], ModificationType.none.name
                 )
-                highest_st_status = get_higher_status(highest_st_status, arr_dep_status)
+
+                # For trip status: ignore back-to-normal delete (stop-time that was not in base-schedule)
+                if stop_event_status == ModificationType.delete.name:
+                    # TODO: be more robust to cases with loops, or where stop order changes in RT.
+                    # Probably do a synchronized pass over realtime and base-schedule VJs
+                    vj_st, _ = get_navitia_stop_time_sncf(
+                        cr=get_value(pdp, "cr"),
+                        ci=get_value(pdp, "ci"),
+                        ch=get_value(pdp, "ch"),
+                        nav_vj=vj.navitia_vj,
+                    )
+                    if not vj_st:
+                        stop_event_status = ModificationType.none.name
```

Maybe I misunderstand the whole thing, but it sounds weird not to be able to find the correct VJ in a case of a delete. 
Are you sure you want to silently ignore this case ?

---

Also, this function is already 150 lines long. I would be much nicer split this logic up in a separate function.

---

True for the function, I will try to do something about it.

The issue is not finding the correct VJ, it's finding the correct **stop-time** (especially in cases where one stop-area appears multiple times in the same trip)

---

For the first comment @nonifier, we discussed this case with the SNCF team, it's the use they want. If a stop is deleted and not in theoritical vj, we consider that nothing happens as it is an added stop that was deleted.